README.md
# Lez in Tink

v0.1.0




## What is this
Proof of concept of the lez project based on the Tink prototype skeleton


## How to run the distribution code (for regular users)

### Prerequisites

Python needs to be installed on your computer. Mac OS X or Linux users have Python installed by default. For windows, please refer to [this page](http://docs.python-guide.org/en/latest/starting/install/win/).


### In your command line

`cd path/to/dist/directory`

`python -m SimpleHTTPServer 8000`


### In your browser

After running the above commands, open your favorite browser and navigate to [http://0.0.0.0:8000](http://0.0.0.0:8000).



## How to contribute (for developers)

### Prerequisites

* nodeJS [http://nodejs.org/download/](http://nodejs.org/download/)
* bower: `npm install -g bower`


### Get started

1. Fork this repo.

2. Run the following in your command line: `npm install`

3. After that, run `npm start`




## Who do I talk to?

* Jasper Van Proeyen - jasper.vanproeyen@digipolis.be - Lead front-end
* Tom Wuyts - tom.wuyts@digipolis.be - Lead UX
* [The hand](https://www.youtube.com/watch?v=_O-QqC9yM28)


CHANGELOG.md
# Changelog Lez in Tink

All notable changes to this project will be documented in this file.
This project adheres to [Semantic Versioning](http://semver.org/).

<!--
## [Unreleased] - [unreleased]

### Added
### Changed
### Deprecated
### Removed
### Fixed
### Security
-->



## [0.1.0] - 2015-09-21

### Added
- Basic setup
- Necessary dirctives


src/scripts/app.js
'use strict';

/**
 * @ngdoc overview
 * @name tinkApp
 * @description
 * # tinkApp
 *
 * Main module of the application.
 */
 angular
 .module('tinkApp', [
   'ngAnimate',
   'ngCookies',
   'ngResource',
   'ngRoute',
   'ngMessages',
   'ngSanitize',
   'ngTouch',
   'tink.navigation',
   'tink.tinkApi',
   'tink.timepicker',
   'tink.datepicker',
   'tink.interactivetable'
   ])
 .config(function ($routeProvider) { /*, $locationProvider */
  $routeProvider
  .when('/', {
    redirectTo: '/alle-aanvragen'
    //controller: 'MainCtrl'
  })
  .when('/alle-aanvragen', {
    templateUrl: 'views/alle-aanvragen.html',
  })
  .when('/alle-hotlists', {
    templateUrl: 'views/alle-hotlists.html',
  })
  .when('/detail-hotlists', {
    templateUrl: 'views/detail-hotlists.html',
  })
  .otherwise({
    templateUrl: 'views/not-implemented.html'
  });
  // $locationProvider.html5Mode({
  //   enabled: true,
  //   requireBase: false
  // });
  // $locationProvider.hashPrefix('!');
});





src/scripts/controllers/main.js
'use strict';

/**
 * @ngdoc function
 * @name tinkApp.controller:MainCtrl
 * @description
 * # MainCtrl
 * Controller of the tinkApp
 */
 angular.module('tinkApp')
 .controller('MainCtrl', function ($scope, $rootScope, $location, $anchorScroll, tinkApi) {


  // Show or hide sidenav
  $scope.$on('$locationChangeStart', function() {
    if($location.path() === '/') {
      // Close the sidenav
      if($scope.sidenav.visible === true) {
        tinkApi.sideNavToggle.closeById('sideNav1');
      }
      // Hide the sidenav toggle
      $scope.sidenav.visible = false;
    } else {
      $scope.sidenav.visible = true;
    }

  });

  // Scope variables
  $scope.sidenav = {
    visible: true
  };

  // Function that replaces anchor scroll
  $scope.scrollTo = function scrollTo(id) {
    $location.hash(id);
    $anchorScroll();
  };

  $scope.openMenu = function(){
    if($location.$$path !== '/') {
      tinkApi.sideNavToggle.openById('sideNav1');
    }
  };

  $scope.closeMenu = function(){
    if($location.$$path !== '/') {
      tinkApi.sideNavToggle.closeById('sideNav1');
    }
  };

});


// 'use strict';

// angular.module('tinkApp')
//  .controller('MainCtrl', function ($scope) {

//   $scope.hello = "Hello!";

// });

src/scripts/controllers/tableHotlistsCtrl.js
'use strict';

angular.module('tinkApp')
  .controller('tableHotlistsCtrl',['$scope','$timeout',function (scope,timeout) {

    scope.data = {};
    // Override default items per page
    scope.pages = '5, 10, 25';

    scope.changed = function(data,next){
      console.log(data,scope.nums);

      timeout(function(){
        // Data model
        scope.data = [
        ];
        scope.totalitems = 200;
        next();
      },300);
    };
    scope.changed = function(type,value,fn){
      fn();
    };

    scope.totalitems = 200;
    scope.chan = function(prop,or,typ){
      console.log('changed',prop,or,typ);
      timeout(function(){
      // Data model
      scope.data = [
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        }
      ];
    },300);
    };

    timeout(function(){
    // Data model
      scope.data.content = [
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
      ];
    },700);

    // The headers you want to show and whether they are already visible in the table
    scope.headers = [
      {
        field: 'datum',
        alias: 'Datum',
        checked: true,
        sort:true
      },
      {
        field: 'wlbl',
        alias: 'WL/BL',
        checked: true,
        sort:true
      },
      {
        field: 'naam',
        alias: 'Naam',
        checked: true,
        sort:true
      },
      {
        field: 'aangemaaktopbo',
        alias: 'Aangemaakt op (BO)',
        checked: true,
        sort:true
      },
      {
        field: 'recordsactief',
        alias: 'Records actief',
        checked: true,
        sort:true
      },
      {
        field: 'recordspassief',
        alias: 'Records passief',
        checked: true,
        sort:true
      },
      {
        field: 'aangemaaktopext',
        alias: 'Aangemaakt op (EXT)',
        checked: true,
        sort:true
      }
    ];
    scope.nums=0;
    scope.changenums = function(){
      scope.nums = scope.nums+1;
    };
    scope.perpageValue=[10,20,25,30,45];


      scope.actions = [
      {
          name: 'remove',
          callback: function(items) {
            angular.forEach(items, function(val) {
              scope.data.content.splice(scope.data.content.indexOf(val),1);
            });
          },
          order:0, //orde of the button
          master:true, //required !
          icon:'fa-close', //the icon required.
          single:true // only when one checkbox is selected
        }
    ];
}]);


src/scripts/controllers/tableDetailHotlistsCtrl.js
'use strict';

angular.module('tinkApp')
  .controller('tableDetailHotlistsCtrl',['$scope','$timeout',function (scope,timeout) {

    scope.data = {};
    // Override default items per page
    scope.pages = '5, 10, 25';

    scope.changed = function(data,next){
      console.log(data,scope.nums);

      timeout(function(){
        // Data model
        scope.data = [
        ];
        scope.totalitems = 200;
        next();
      },300);
    };
    scope.changed = function(type,value,fn){
      fn();
    };

    scope.totalitems = 200;
    scope.chan = function(prop,or,typ){
      console.log('changed',prop,or,typ);
      timeout(function(){
      // Data model
      scope.data = [
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        },
        {
          datum: '21/08/2015',
          wlbl: 'Whitelist',
          naam: 'LNE',
          aangemaaktopbo: '10/09/2015 - 23:02:06',
          recordsactief: '8010000',
          recordspassief: '2',
          aangemaaktopext: '10/09/2015 - 21:08:26'
        }
      ];
    },300);
    };

    timeout(function(){
    // Data model
      scope.data.content = [
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'false',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'false',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Chantal De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'false',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        },
        {
          nummerplaat: '1-AZE-567',
          land: 'België',
          actief: 'true',
          gewijzigdop: '10/09/2015 - 23:02:06',
          gewijzigddoor: 'Luc De Backer',
          reden: 'Correctie programmafout'
        }
      ];
    },700);

    // The headers you want to show and whether they are already visible in the table
    scope.headers = [
      {
        field: 'nummerplaat',
        alias: 'Nummerplaat',
        checked: true,
        sort:true
      },
      {
        field: 'land',
        alias: 'Land',
        checked: true,
        sort:true
      },
      {
        field: 'actief',
        alias: 'Actief',
        checked: true,
        sort:true
      },
      {
        field: 'gewijzigdop',
        alias: 'Gewijzigd op',
        checked: true,
        sort:true
      },
      {
        field: 'gewijzigddoor',
        alias: 'Gewijzigd door',
        checked: true,
        sort:true
      },
      {
        field: 'reden',
        alias: 'Reden',
        checked: true,
        sort:true
      }
    ];
    scope.nums=0;
    scope.changenums = function(){
      scope.nums = scope.nums+1;
    };
    scope.perpageValue=[10,20,25,30,45];


      scope.actions = [
      {
          name: 'Detaillijn wijzigen',
          callback: function(items) {
            angular.forEach(items, function(val) {
              scope.data.content.splice(scope.data.content.indexOf(val),1);
            });
          },
          order:0, //orde of the button
          master:true, //required !
          icon:'fa-check', //the icon required.
          single:true // only when one checkbox is selected
        }
    ];
}]);


src/scripts/controllers/tableCtrl.js
'use strict';

angular.module('tinkApp')
  .controller('tableCtrl',['$scope','$timeout',function (scope,timeout) {

    scope.data = {};
    // Override default items per page
    scope.pages = '5, 10, 25';

    scope.changed = function(data,next){
      console.log(data,scope.nums);

      timeout(function(){
        // Data model
        scope.data = [
        ];
        scope.totalitems = 200;
        next();
      },300);
    };
    scope.changed = function(type,value,fn){
      fn();
    };

    scope.totalitems = 200;
    scope.chan = function(prop,or,typ){
      console.log('changed',prop,or,typ);
      timeout(function(){
      // Data model
      scope.data = [
        {
          aanvraagnr: '2015.34034',
          nummerplaat: '1 - AAA - 123',
          aangevraagdop: '15/02/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 1',
          looptijdstart: '15/02/2015',
          looptijdeinde: '10/03/2015',
        },
        {
          aanvraagnr: '2015.94893',
          nummerplaat: '1 - OAJ - 123',
          aangevraagdop: '23/09/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Oldtimer',
          looptijdstart: '23/09/2015',
          looptijdeinde: '25/09/2015',
        },
        {
          aanvraagnr: '2015.19847',
          nummerplaat: '1 - TXA - 123',
          aangevraagdop: '12/06/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 1',
          looptijdstart: '12/06/2015',
          looptijdeinde: '12/06/2015',
        },
        {
          aanvraagnr: '2015.40392',
          nummerplaat: '1 - DDD - 123',
          aangevraagdop: '11/02/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 1',
          looptijdstart: '15/02/2015',
          looptijdeinde: '10/03/2015',
        }
      ];
    },300);
    };

    timeout(function(){
    // Data model
      scope.data.content = [
        {
          aanvraagnr: '2015.34034',
          nummerplaat: '1 - AAA - 123',
          aangevraagdop: '15/02/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 1',
          looptijdstart: '15/02/2015',
          looptijdeinde: '10/03/2015',
        },
        {
          aanvraagnr: '2015.94893',
          nummerplaat: '1 - OAJ - 123',
          aangevraagdop: '23/09/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Oldtimer',
          looptijdstart: '23/09/2015',
          looptijdeinde: '25/09/2015',
        },
        {
          aanvraagnr: '2015.19847',
          nummerplaat: '1 - TXA - 123',
          aangevraagdop: '12/06/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 1',
          looptijdstart: '12/06/2015',
          looptijdeinde: '12/06/2015',
        },
        {
          aanvraagnr: '2015.40392',
          nummerplaat: '1 - DDD - 123',
          aangevraagdop: '11/02/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 1',
          looptijdstart: '15/02/2015',
          looptijdeinde: '10/03/2015',
        },
        {
          aanvraagnr: '2015.98763',
          nummerplaat: '1 - FDS - 123',
          aangevraagdop: '01/04/2015',
          status: 'ingediend',
          type: 'Vrijstelling',
          soort: 'Soort 1',
          looptijdstart: '01/04/2015',
          looptijdeinde: '19/06/2015',
        },
        {
          aanvraagnr: '2015.45678',
          nummerplaat: '1 - AZE - 123',
          aangevraagdop: '10/02/2015',
          status: 'opnieuw ingediend',
          type: 'Vrijstelling',
          soort: 'Oldtimer',
          looptijdstart: '10/02/2015',
          looptijdeinde: '28/03/2015',
        },
        {
          aanvraagnr: '2015.36738',
          nummerplaat: '1 - EDR - 123',
          aangevraagdop: '06/07/2015',
          status: 'afgekeurd',
          type: 'Vrijstelling',
          soort: 'Soort 4',
          looptijdstart: '06/07/2015',
          looptijdeinde: '17/09/2015',
        },
        {
          aanvraagnr: '2015.76378',
          nummerplaat: '1 - ABC - 123',
          aangevraagdop: '19/04/2015',
          status: 'goedgekeurd',
          type: 'Vrijstelling',
          soort: 'Soort 4',
          looptijdstart: '19/04/2015',
          looptijdeinde: '10/06/2015',
        },
        {
          aanvraagnr: '2015.87639',
          nummerplaat: '1 - JHZ - 123',
          aangevraagdop: '21/04/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 7',
          looptijdstart: '21/04/2015',
          looptijdeinde: '23/06/2015',
        },
        {
          aanvraagnr: '2015.95673',
          nummerplaat: '1 - KSH - 123',
          aangevraagdop: '09/03/2015',
          status: 'goedgekeurd',
          type: 'Vrijstelling',
          soort: 'Soort 4',
          looptijdstart: '09/03/2015',
          looptijdeinde: '13/05/2015',
        },
        {
          aanvraagnr: '2015.34563',
          nummerplaat: '1 - FSG - 123',
          aangevraagdop: '16/05/2015',
          status: 'in behandeling',
          type: 'Vrijstelling',
          soort: 'Soort 4',
          looptijdstart: '16/05/2015',
          looptijdeinde: '17/08/2015',
        },
        {
          aanvraagnr: '2015.34567',
          nummerplaat: '1 - SJS - 123',
          aangevraagdop: '31/01/2015',
          status: 'goedgekeurd',
          type: 'Vrijstelling',
          soort: 'Oldtimer',
          looptijdstart: '31/01/2015',
          looptijdeinde: '13/03/2015',
        },
      ];
    },700);

    // The headers you want to show and whether they are already visible in the table
    scope.headers = [
      {
        field: 'aanvraagnr',
        alias: 'Aanvraagnr',
        checked: true,
        sort:true
      },
      {
        field: 'nummerplaat',
        alias: 'Nummerplaat',
        checked: true,
        sort:true
      },
      {
        field: 'aangevraagdop',
        alias: 'Aangevraagd op',
        checked: true,
        sort:true
      },
      {
        field: 'status',
        alias: 'Status',
        checked: true,
        sort:true
      },
      {
        field: 'type',
        alias: 'Type',
        checked: true,
        sort:true
      },
      {
        field: 'soort',
        alias: 'Soort',
        checked: true,
        sort:true
      },
      {
        field: 'looptijdstart',
        alias: 'Looptijd start',
        checked: true,
        sort:true
      },
      {
        field: 'looptijdeinde',
        alias: 'Looptijd einde',
        checked: true,
        sort:true
      }
    ];
    scope.nums=0;
    scope.changenums = function(){
      scope.nums = scope.nums+1;
    };
    scope.perpageValue=[10,20,25,30,45];


      scope.actions = [
      {
          name: 'remove',
          callback: function(items) {
            angular.forEach(items, function(val) {
              scope.data.content.splice(scope.data.content.indexOf(val),1);
            });
          },
          order:0, //orde of the button
          master:true, //required !
          icon:'fa-close', //the icon required.
          single:true // only when one checkbox is selected
        }
    ];
}]);
